crates/joystick/src/linux.rs

```diff
@@ -28,139 +23,184 @@ pub struct Joystick {
     instant: Instant,
 }
 
-impl Default for Joystick {
-    fn default() -> Self {
-        let name = "Future TAS Joystick Linux";
-        let abs_setup = AbsInfo::new(256, 0, 512, 20, 20, 1);
-        let abs_x = UinputAbsSetup::new(AbsoluteAxisCode::ABS_X, abs_setup);
-        let abs_y = UinputAbsSetup::new(AbsoluteAxisCode::ABS_Y, abs_setup);
-        let mut keys = AttributeSet::<KeyCode>::new();
+impl JoystickInterface for Joystick {
+    fn press_a(&mut self) {
+        self.press(KeyCode::BTN_EAST)
```

Comment:
Are these correct? At least on my Xbox360 controller, the layout is:
```
  Y
X   B
  A
```

Reply:
That will need testing - linux reverses the axis i think